ClockDisplay.java

```diff
@@ -35,12 +35,24 @@ public ClockDisplay()
      * creates a new clock set at the time specified by the 
      * parameters.
      */
-    public ClockDisplay(int hour, int minute, String meridian)
+    public ClockDisplay(int hour, int minute, String newMeridian)
     {
-        hours = new NumberDisplay(12);
-        minutes = new NumberDisplay(60);
-        this.meridian = meridian;
-        setTime(hour, minute, meridian);
+        if(hour>0 && hour<=12)
+        {
+            if(newMeridian.equalsIgnoreCase("AM") || newMeridian.equalsIgnoreCase("PM"))
+            {
+                hours.setValue(hour);
+                minutes.setValue(minute);
+                meridian = newMeridian;
+                updateDisplay();
+            }
+            else{
+                System.out.println("Meridian value invalid. Please use am or pm.");
```

Comment:
and this is why you don't want to use strings, but I am glad that you checked for and caught possible errors